core/trino-parser/src/main/java/io/trino/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -824,16 +824,18 @@ protected Expression visitRowDataType(RowDataType node, Context<C> context)
 
                 Optional<Identifier> name = field.getName();
 
+                boolean nameRewritten = false;
                 if (field.getName().isPresent()) {
                     Identifier identifier = field.getName().get();
                     Identifier rewrittenIdentifier = rewrite(identifier, context.get());
 
                     if (identifier != rewrittenIdentifier) {
+                        nameRewritten = true;
                         name = Optional.of(rewrittenIdentifier);
                     }
                 }
 
-                if (dataType != field.getType() || name != field.getName()) {
+                if (dataType != field.getType() || nameRewritten) {
```

Comment:
This seems less clear than before since dataType is being compared the same way here.

If we really want this errorprone, maybe just add the suppression @SuppressWarnings("OptionalEquality") here so that this method still looks like all the other methods.

Reply:
I agree. This is another case of error prone rules getting in the way of perfectly reasonable code and forcing us to jump through hoops to make it happy.

Reply:
I do not agree.
I was starring at this code and was really unsure whether this is intentional, or a mistake.
Only after i realized the Optional is recreated conditionally ealier, i determined this is not a bug.
So errorprone aside, I insist that my change here _is an improvement_.

And yet, this is the only disputable occurrence of the problem.
During development when retrofitting something from `@Nullable` to `Optional`, the check can come really help to help you find those ` != null` cases.
I don't believe we have 100% test coverage, so this is really useful.

Reply:
I agree this could be a useful check in some cases, but in this context I think it makes more sense to retain == since that is how every other field in this visitor is being compared -- to me that consistency across all N number of visit methods is more important here.

As you pointed out, this is the only case where the check made a difference, so if that is any indicator, this is probably not going to be a commonly firing check one way or another. In my mind, seeing the suppression is also sufficient documentation for future reference.